sdk/metricsadvisor/azure-ai-metricsadvisor/src/test/java/com/azure/ai/metricsadvisor/AnomalyAlertAsyncTest.java

```diff
@@ -160,20 +162,24 @@ public void getAnomalyAlertValidId(HttpClient httpClient, MetricsAdvisorServiceV
     @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
     @MethodSource("com.azure.ai.metricsadvisor.TestUtils#getTestParameters")
     public void createAnomalyAlertConfiguration(HttpClient httpClient, MetricsAdvisorServiceVersion serviceVersion) {
-        // Arrange
-        client = getMetricsAdvisorAdministrationBuilder(httpClient, serviceVersion).buildAsyncClient();
         final AtomicReference<String> alertConfigurationId = new AtomicReference<>();
-        creatAnomalyAlertRunner(inputAnomalyAlert ->
-
-            // Act & Assert
-            StepVerifier.create(client.createAnomalyAlertConfiguration(inputAnomalyAlert))
-                .assertNext(createdAnomalyAlert -> {
-                    alertConfigurationId.set(createdAnomalyAlert.getId());
-                    validateAnomalyAlertResult(inputAnomalyAlert, createdAnomalyAlert);
-                })
-                .verifyComplete());
-
-        client.deleteAnomalyAlertConfiguration(alertConfigurationId.get()).block();
+        try {
+           // Arrange
+            client = getMetricsAdvisorAdministrationBuilder(httpClient, serviceVersion).buildAsyncClient();
+            creatAnomalyAlertRunner(inputAnomalyAlert ->
+
+                // Act & Assert
+                StepVerifier.create(client.createAnomalyAlertConfiguration(inputAnomalyAlert))
+                    .assertNext(createdAnomalyAlert -> {
+                        alertConfigurationId.set(createdAnomalyAlert.getId());
+                        validateAnomalyAlertResult(inputAnomalyAlert, createdAnomalyAlert);
+                    })
+                    .verifyComplete());
+        } finally {
+            Mono<Void> deleteAnomalyAlertConfiguration
+                = client.deleteAnomalyAlertConfiguration(alertConfigurationId.get());
+            StepVerifier.create(deleteAnomalyAlertConfiguration).verifyComplete();
```

Comment:
In `finally` should we attempt to clean up the resource only if the `alertConfigurationId` has a value (which means creation succeeded).

Reply:
the same comment for other applicable places we do the cleanup in finally.